### PR TITLE
fix(cli): handle empty files

### PR DESCRIPTION
**Pull request by jiparis**
Gracefully handle empty files uploads.

Closes https://github.com/chainloop-dev/chainloop/issues/380

### CODE REVIEW

File: app/cli/internal/action/artifact_upload.go [hunk 1]
```diff
@@ -59,6 +59,11 @@ func (a *ArtifactUpload) Run(filePath string) (*CASArtifact, error) {
 		return nil, fmt.Errorf("retrieving file information: %w", err)
 	}
 
+	if info.Size() == 0 {
+		a.Logger.Info().Msg("the file is empty, nothing to upload")
```
Comment (buccarel) [inline app/cli/internal/action/artifact_upload.go:63]: I have a context question, most likely for @migmartri, but anybody who knows the answer is welcome to reply ofc.

What is the context for these files?
Would someone find legit, in the generic scenario, to produce and upload an empty file?

Comment (migmartri) [reply app/cli/internal/action/artifact_upload.go:63]: Interesting point.

I think you are right, it doesn't harm to upload empty files. In fact, artifact material types in attestations can be empty, but this code doesn't affect that (this is only for `chainloop artifact upload` cmd.) 

The change in `uploader.go` is in fact what fixes the original panic #430, so I think it's ok if we remove 

```go
if info.Size() == 0 {
		a.Logger.Info().Msg("the file is empty, nothing to upload")
...
```

and just keep the change at `internal/casclient/uploader.go`.

What do you think?

> Note: This implementation is based on my prescription offline, so any blame is on me :)

**Review by migmartri — COMMENTED** [review]
Thanks for the contribution! 

Adding some comments to reply to @buccarel

**Review by migmartri — APPROVED** [review]

File: app/cli/internal/action/artifact_upload.go (hunk 1 above)
Comment (jiparis) [reply app/cli/internal/action/artifact_upload.go:63]: Makes sense. I'll update the PR with your sugested changes. Thanks!

Comment (buccarel) [reply app/cli/internal/action/artifact_upload.go]: Ok perfect. I'll request changes to remember not to merge this until it's fixed. 
Thanks.

**Review by buccarel — CHANGES_REQUESTED** [review]
Waiting for the fix we agreed on

**Review by migmartri — APPROVED** [review]
LGTM! Thanks!

**Comment by migmartri** [comment]
> Waiting for the fix we agreed on

@buccarel ptal, it seems that the fix is now up.

**Pull request merged**